lms/djangoapps/course_goals/tests/test_api.py

```diff
@@ -32,21 +33,22 @@ def setUp(self):
         self.course = CourseFactory.create(emit_signals=True)
 
         self.user = User.objects.create_user('john', 'lennon@thebeatles.com', 'password')
-        CourseEnrollment.enroll(self.user, self.course.id)
+        self.enrollment = CourseEnrollment.enroll(self.user, self.course.id)
 
         self.client = APIClient(enforce_csrf_checks=True)
         self.client.login(username=self.user.username, password=self.user.password)
         self.client.force_authenticate(user=self.user)
 
         self.apiUrl = reverse('course_goals_api:v0:course_goal-list')
 
-    @mock.patch('lms.djangoapps.course_goals.views.segment.track')
+    @mock.patch('lms.djangoapps.course_goals.handlers.segment.track')
     @override_settings(LMS_SEGMENT_KEY="foobar")
-    def test_add_valid_goal(self, ga_call):
+    def test_add_valid_goal(self, segment_call):
         """ Ensures a correctly formatted post succeeds."""
         response = self.post_course_goal(valid=True, goal_key='certify')
-        assert self.get_event(- 1)['name'] == EVENT_NAME_ADDED
```

Comment:
These calls added no value because the assert called right below validates the name of the event as well.